### PR TITLE
Refactor config for Vue client webpack

### LINKED ISSUE #15403: Cleanup vue webpack config.
##### **Overview of the issue**

While doing some changes to vue webpack at https://github.com/jhipster/generator-jhipster/pull/15210 and failing, discovered that vue webpack config needs a cleanup.
- Angular config uses 2 files, react uses 5, while vue uses 9.
- Some configs are never used and are misleading, eg:
https://github.com/jhipster/generator-jhipster/blob/1e42b520cdc8328cc7c96c09c9ecae65824f049d/generators/client/templates/vue/webpack/env.js.ejs#L15-L16
- Some implementations are wrong.
process.env.NODE_ENV is never set at this point.
https://github.com/jhipster/generator-jhipster/blob/1e42b520cdc8328cc7c96c09c9ecae65824f049d/generators/client/templates/vue/webpack/loader.conf.js.ejs#L4
- Use of process.env should be avoided https://github.com/webpack/changelog-v5/blob/master/MIGRATION%20GUIDE.md#level-5-runtime-errors, but some lib still requires it, need to track.
https://github.com/jhipster/generator-jhipster/blob/1e42b520cdc8328cc7c96c09c9ecae65824f049d/generators/client/templates/vue/webpack/webpack.common.js.ejs#L123

I have a partial implementation splited from https://github.com/jhipster/generator-jhipster/pull/15210, but it needs more work.

##### **Motivation for or Use Case**

Current config files are complicated and changes doesn't applies.

##### **Related issues**

Related to https://github.com/jhipster/generator-jhipster/pull/15210 and https://github.com/jhipster/generator-jhipster/pull/15107

##### **Browsers and Operating System**

- [ ] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by shershen08**
Seems `path.resolve` will do the job, so removed utils.js first
Also planning to mv `loader.conf.js` inside `webpack.common.js`

---

Please make sure the below checklist is followed for Pull Requests.

- [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [x] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

Fixes #15403

### CODE REVIEW

File: generators/client/templates/vue/webpack/webpack.dev.js.ejs [hunk 1]
```diff
@@ -24,7 +24,7 @@ module.exports = webpackMerge(baseWebpackConfig, {
     main: './<%= MAIN_SRC_DIR %>app/main'
   },
   output: {
-    path: jhiUtils.root('<%= DIST_DIR %>'),
+    path: path.resolve('<%= DIST_DIR %>'),
```
Comment (mshima) [inline generators/client/templates/vue/webpack/webpack.dev.js.ejs:27]:
```suggestion
    path: path.resolve(__dirname, '..', '<%= DIST_DIR %>'),
```
Not sure but without this npm start could fail if you are not at the project root.

**Review by mshima — COMMENTED** [review]
You may move output.path to webpack.common.

**Comment by mshima** [comment]
This isn’t enough to fix https://github.com/jhipster/generator-jhipster/issues/15403.

File: generators/client/templates/vue/webpack/webpack.dev.js.ejs (hunk 1 above)
Comment (shershen08) [reply generators/client/templates/vue/webpack/webpack.dev.js.ejs]: did that

**Comment by shershen08** [comment]
> This isn’t enough to fix #15403.

webpack/env.js could also be moved inside `webpack.common.js`, but it will be less readable

**Comment by shershen08** [comment]
this issue was closed by some reason - https://github.com/jhipster/generator-jhipster/issues/15403 
what else might be missing in this PR @mshima ?

**Comment by DanielFran** [comment]
@shershen08 It was closed automatically, I reopen it

**Comment by mshima** [comment]
> this issue was closed by some reason - #15403
> what else might be missing in this PR @mshima ?

I didn’t realized before, but your this branch is based on an year old commit.

Not sure if you want to rebase and finish this or help with https://github.com/jhipster/generator-jhipster/pull/15932.
css is not loading correctly, maybe a bug in vue-cli so I decided to wait.

**Comment by shershen08** [comment]
> > this issue was closed by some reason - #15403
> > what else might be missing in this PR @mshima ?
> 
> I didn’t realized before, but your this branch is based on an year old commit.
> 
> Not sure if you want to rebase and finish this or help with #15932.
> css is not loading correctly, maybe a bug in vue-cli so I decided to wait.

Oh definitely I was thinking about Vue-cli as more future proof solution. Just haven't seen this issue. I will then try to incorporate altogether

**Pull request closed without merging**